packages/protocol/contracts/L1/access/IProposerAccess.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+/// @title IProposerAccess
+/// @dev An interface to check if a proposer is eligible to propose blocks in a specific Ethereum
+/// block.
+/// @custom:security-contact security@taiko.xyz
+interface IProposerAccess {
+    /// @notice Checks if a proposer can propose block in the current Ethereum block.
+    /// @param _proposer The proposer.
+    /// @return true if the proposer can propose blocks, false otherwise.
+    function isProposerEligible(address _proposer) external view returns (bool);
+}
```

Comment:
Just for client interface point of view, would be good to use the exact same namings like [here](https://github.com/taikoxyz/taiko-mono/blob/015ed3499f28002f42e27164812bfd4c077f7c0b/packages/protocol/contracts/L1/ISequencerRegistry.sol), in [this](https://github.com/taikoxyz/taiko-mono/pull/17654) PR,  if we wanna reuse the backend, OR use this naming there.. But definitely would be cool to avoid duplicated work later.

Reply:
I don't like the word `sequencer`. And I also believe going forward, ISequencerRegistry will also change due to the dependency on restaking platforms/protocols.

Reply:
`ISequencerRegistry` very likely stable because how the implementation selects the current proposer does not matter to taiko.

Sequencer vs proposer is subjective, I'm fine with both but people do tend to use the word sequencer here to stress that they are sequencing the blocks and there's only one.

Reply:
We can change it later. I think we just care about the permission checking function rather than the registration logics.